cli/src/main/java/org/jboss/as/cli/batch/impl/DefaultBatch.java

```diff
@@ -103,8 +106,25 @@ public ModelNode toRequest() {
         composite.get(Util.OPERATION).set(Util.COMPOSITE);
         composite.get(Util.ADDRESS).setEmptyList();
         final ModelNode steps = composite.get(Util.STEPS);
-        for(BatchedCommand cmd : commands) {
-            steps.add(cmd.getRequest());
+        for (BatchedCommand cmd : commands) {
+            CommandContext ctx = cmd.getCommandContext();
+            ModelNode request = cmd.getRequest();
+            if (ctx.getConfig().isValidateOperationRequests()) {
+                try {
+                    ctx.set(Scope.REQUEST, Util.DESCRIPTION_RESPONSE,
+                            cmd.getDescriptionResponse());
+                    ModelNode opDescOutcome = Util.validateRequest(ctx, request);
+                    if (opDescOutcome != null) { // operation has params that might need to be replaced
+                        Util.replaceFilePathsWithBytes(request, opDescOutcome);
```

Comment:
The description is only retrieved when validation is enabled, can we really move it outside validation? It would mean the the would make the description retrieval in all cases.